mypy/nodes.py

```diff
@@ -2133,7 +2128,7 @@ def calculate_mro(self) -> None:
         self.mro = mro
         # The property of falling back to Any is inherited.
         self.fallback_to_any = any(baseinfo.fallback_to_any for baseinfo in self.mro)
-        self.reset_subtype_cache()
+        TypeState.reset_all_subtype_caches_for(self)
```

Comment:
Bummer that we still have the circular dependency. We could move the `calculate_mro` method to another module (that `mypy.node` doesnt't depend on) to get rid of it. Can you do it or at least add a TODO comment about this refactor?